pkg/rc/replication_controller.go

```diff
@@ -141,72 +157,129 @@ func (rc *replicationController) meetDesires() error {
 			return err
 		}
+	}
+	return nil
+}
 
-		// If we need to downsize the number of nodes, prefer any in current that are not eligible anymore.
-		// TODO: evaluate changes to 'eligible' more frequently
-		preferred := sets.NewString(current...).Difference(sets.NewString(eligible...))
-		rest := sets.NewString(current...).Difference(preferred)
-		toUnschedule := len(current) - rc.ReplicasDesired
-		rc.logger.NoFields().Infof("Need to unschedule %d nodes out of %s", toUnschedule, current)
+func (rc *replicationController) removePods(current PodLocations) error {
+	currentNodes := current.Nodes()
+	eligible, err := rc.eligibleNodes()
+	if err != nil {
+		return err
+	}
 
-		for i := 0; i < toUnschedule; i++ {
-			unscheduleFrom, ok := preferred.PopAny()
+	// If we need to downsize the number of nodes, prefer any in current that are not eligible anymore.
+	// TODO: evaluate changes to 'eligible' more frequently
+	preferred := sets.NewString(currentNodes...).Difference(sets.NewString(eligible...))
+	rest := sets.NewString(currentNodes...).Difference(preferred)
+	toUnschedule := len(current) - rc.ReplicasDesired
+	rc.logger.NoFields().Infof("Need to unschedule %d nodes out of %s", toUnschedule, current)
+
+	for i := 0; i < toUnschedule; i++ {
+		unscheduleFrom, ok := preferred.PopAny()
+		if !ok {
+			var ok bool
+			unscheduleFrom, ok = rest.PopAny()
 			if !ok {
-				var ok bool
-				unscheduleFrom, ok = rest.PopAny()
-				if !ok {
-					// This should be mathematically impossible unless replicasDesired was negative
-					return util.Errorf(
-						"Unable to unschedule enough nodes to meet replicas desired: %d replicas desired, %d current.",
-						rc.ReplicasDesired, len(current),
-					)
-				}
+				// This should be mathematically impossible unless replicasDesired was negative
+				return util.Errorf(
+					"Unable to unschedule enough nodes to meet replicas desired: %d replicas desired, %d current.",
+					rc.ReplicasDesired, len(current),
+				)
 			}
-			err := rc.unschedule(unscheduleFrom)
+		}
+		err := rc.unschedule(unscheduleFrom)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (rc *replicationController) ensureConsistency(current PodLocations) error {
+	manifestSHA, err := rc.Manifest.SHA()
+	if err != nil {
+		return err
+	}
+	for _, pod := range current {
+		intent, _, err := rc.kpStore.Pod(kp.INTENT_TREE, pod.Node, types.PodID(pod.PodID))
+		if err != nil && err != pods.NoCurrentManifest {
+			return err
+		}
+		var intentSHA string
+		if intent != nil {
+			intentSHA, err = intent.SHA()
 			if err != nil {
-				return err
+				rc.logger.WithError(err).WithField("node", pod.Node).Warn("Could not hash manifest to determine consistency of intent")
+			}
+			if intentSHA == manifestSHA {
+				continue
 			}
 		}
-	} else {
-		rc.logger.NoFields().Debugln("Taking no action")
+
+		rc.logger.WithField("node", pod.Node).WithField("intentManifestSHA", intentSHA).Info("Found inconsistency in scheduled manifest")
```

Comment:
I agree that `WithFields()` would work too, but since you point this out a day after the PR was merged, I'm reluctant to change it simply for stylistic reasons.